File: Nov2020/28Nov/FrequencyOfStudents.cpp
/*
Problem Name: Frequency of Students
Problem Link: https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/practice-problems/algorithm/frequency-of-students/
*/

#include<bits/stdc++.h>
using namespace std;
int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    int n;
    cin>>n;
    string s[n];
    map<string,int> m;
    for(int i=0;i<n;++i){
        cin>>s[i];
        m[s[i]]++;
    }
    sort(s,s+n);
    for(int i=0;i<n;++i){
        if(m[s[i]]>0)
            cout<<s[i]<<" "<<m[s[i]]<<"\n";
            m[s[i]]=-1;
    }
    return 0;
}

File: Nov2020/28Nov/FrequencyOfStudents.java
/*
Problem Name: Frequency of Students
Problem Link: https://www.hackerearth.com/practice/data-structures/hash-tables/basics-of-hash-tables/practice-problems/algorithm/frequency-of-students/
*/
import java.util.*;
class FrequencyOfStudents {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        TreeMap<String,Integer> map = new TreeMap<String,Integer>();
        int n = in.nextInt();
        for(;n>0;n--){
            String name = in.next();
            if(map.containsKey(name))
                map.put(name, map.get(name) + 1);
            else
                map.put(name, 1);
        }
        for(String name: map.keySet())
            System.out.println(name+" "+map.get(name));
    }
}
